### PR TITLE
Fixup simple memory eviction

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Fixes -- https://github.com/kubernetes/kubernetes/pull/31939

Marking do-not-merge for now, but @vishh @Random-Liu @mtaufen this was passing consistently for me.  Had thrown some logging in eviction manager while debugging as well.

Going back to looking at the other eviction test.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/32006)

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
@aveshagarwal ^^ -- let me know if this passes for you as well on Monday.

**Comment by derekwaynecarr** [comment]
To be clear, the issue I saw in the source test was that it failed deleting the guaranteed pod cause the pod was already evicted in the previous test, but then deleted as part of namespace clean-up in previous It() block.

**Comment by derekwaynecarr** [comment]
@k8s-bot test this https://github.com/kubernetes/kubernetes/issues/27462

File: test/e2e_node/memory_qos_eviction_test.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
+/*
```
Comment (mtaufen) [inline test/e2e_node/memory_qos_eviction_test.go]: Will want to rebase on #31912 which made some changes to this before the file rename.

Comment (derekwaynecarr) [inline test/e2e_node/memory_qos_eviction_test.go]: I don't even want to merge this!  I just put it up for comparison right now as I debug the original PR.

**Comment by derekwaynecarr** [comment]
Looks like git fetch was broken on PR builders...

**Comment by mtaufen** [comment]
That's... interesting.

Ack as far as the attempt to delete across namespaces. That makes sense. Now we just have to figure out why there is an interval during which the eviction manager refuses to admit new pods after the pressure condition appears to disappear from the node.

**Comment by derekwaynecarr** [comment]
@mtaufen - this PR for me locally did not demonstrate that problem, but its also gating the admit check behind another It() block so there is namespace delete/create flow before the besteffort pod tries to come-in.  Going to restructure this test to remove that to see if it repros the core issue.

**Comment by derekwaynecarr** [comment]
@mtaufen @vishh --- i restructured this test to try and use it as a more accurate reproduction.  i cannot get this test to fail for me now locally after many runs, and the best-effort pod creation occurs right after memory pressure is relieved.  PTAL if you see same locally.

File: test/e2e_node/memory_eviction_test.go [hunk 2]
```diff
@@ -93,11 +99,6 @@ var _ = framework.KubeDescribe("MemoryEvictionSimple [FLAKY] [Slow] [Serial] [Di
 			}, 5*time.Minute, 5*time.Second).Should(BeNil())
 		})
 		It("should drop the node memory pressure condition", func() {
-			By("deleting the memory hog pod.")
```
Comment (vishh) [inline test/e2e_node/memory_eviction_test.go]: Why not delete the pod? Deletion should be a no-op right even if the pod got evicted?

File: pkg/kubelet/eviction/eviction_manager.go [hunk 3]
```diff
@@ -94,22 +94,30 @@ func NewManager(
 func (m *managerImpl) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
 	m.RLock()
 	defer m.RUnlock()
+	defer glog.Infof("eviction manager: end Admit: %v", attrs.Pod.Name)
+	glog.Infof("eviction manager: start Admit: %v", attrs.Pod.Name)
```
Comment (vishh) [inline pkg/kubelet/eviction/eviction_manager.go]: Do we want these logs to be Info?

**Comment by vishh** [comment]
@mtaufen can you give this test a try locally?

File: pkg/kubelet/eviction/eviction_manager.go (hunk 3 above)
Comment (derekwaynecarr) [inline pkg/kubelet/eviction/eviction_manager.go]: I don't want any of these logs right now, it was purely for debugging right now locally to isolate that admit was not the problem.

**Comment by derekwaynecarr** [comment]
@vishh -- basically, i am trying to show the problem from what i can tell is not a race w/ transition period as we should have seen that with higher eviction thresholds.  wlil pick this up monday.

**Comment by mtaufen** [comment]
Passes with 60% memory consumption limit (which is what is set in `createMemhogPod` in this PR, see line 192 of `memory_qos_eviction_test.go`), but I end up getting `Ginkgo timed out waiting for all parallel nodes to report back!` when I try to run it with 80% consumption.

**Comment by derekwaynecarr** [comment]
Confused on parallel as I had run this with parallelism of 1.  Just so I
understand you are running test on 8?? different nodes when not set to 1?

**Comment by mtaufen** [comment]
My superficial understanding is that it runs across 8 by default. I can take a deeper look at this.

**Comment by mtaufen** [comment]
Note: I am running via `make test-e2e-node FOCUS=MemoryEviction REMOTE=true`, what are you doing?

**Comment by mtaufen** [comment]
Yeah `parallelism=${PARALLELISM:-8}` from `test-e2e-node.sh`. I can try it with 80% and parallelism of 1, but ideally this should work with the default setting of 8.

**Comment by mtaufen** [comment]
`make test-e2e-node FOCUS=MemoryEvictionSimple REMOTE=true PARALLELISM=1` passes for me at 70%, but still fails for me at 80%.

**Comment by vishh** [comment]
Ping. Can we get this test merged?

**Comment by mtaufen** [comment]
Before merging this, make sure `memory_qos_eviction_test.go` lines up with the latest upstream `memory_eviction_test.go`. I would also prefer if we could clean up some of the file/test naming, my suggestion is this:
- `memory_eviction_simple.go` for `MemoryEvictionSimple`. But we really ought to do something more descriptlive, like: `memory_eviction_admit_after_evict.go`/`MemoryEvictionAdmitAfterEvict`, since it tests whether a pod can be admitted following a previous eviction. 
- `memory_eviction_qos.go` for what was `MemoryEviction` but should be called `MemoryEvictionQOS` moving forward

Also, FYI, once https://github.com/kubernetes/kubernetes/pull/32530 goes in I'm going to convert the memory eviction test to use those utility functions for dynamic configuration. These utilities might also help with `TODO: Fetch kubelet configuration and set this value based on eviction thresholds.`, which is a TODO introduced in this PR.

I also noticed several all-caps log messages in this PR, those might deserve some cleanup as well.

**Comment by derekwaynecarr** [comment]
@vishh - had many distractions, will get back to this this week.

**Comment by derekwaynecarr** [comment]
@mtaufen -- i started to do clean up as requested, but ran out of time to finish today.  either way, i would like to hold this pr until https://github.com/kubernetes/kubernetes/pull/32530 so we can do the dynamic config pieces.

**Comment by mtaufen** [comment]
@derekwaynecarr just got back around to this, I'll rebase #32530 today.

**Comment by mtaufen** [comment]
Rebased #32530, pending review.

**Comment by vishh** [comment]
@derekwaynecarr @mtaufen is this ready for review?

**Comment by derekwaynecarr** [comment]
I will update this week again.

**Comment by fejta** [comment]
@derekwaynecarr please reopen this and/or create a new PR when this is ready for review.

**Pull request closed without merging**